-feat:

ArsenyIshtubaev opened the pull request:
Спринт9 (1)
- улучшить API приложения до соответствия REST
- изменить архитектуру приложения с помощью внедрения зависимостей

ArsenyIshtubaev committed:
-feat:
Спринт9 (1)
- улучшить API приложения до соответствия REST
- изменить архитектуру приложения с помощью внедрения зависимостей

DyndaNE commented on src/main/java/ru/yandex/practicum/filmorate/controller/FilmController.java:
```diff
@@ -1,67 +1,62 @@
 
-    private final Set<Film> films = new HashSet<>();
+    @Autowired
+    private FilmService filmService;
```
тебе стоит почитать про внедрение зависимостей через конструкторы, в данный момент применено неправильно

DyndaNE commented on src/main/java/ru/yandex/practicum/filmorate/controller/FilmController.java:
```diff
@@ -1,67 +1,62 @@
-    public Set<Film> findAll() {
-        return films;
+    public List<Film> findAll() {
+        return filmService.getFilmStorage().findAll();
```
твой контроллер должен обращаться и вызывать методы сервиса, а сервис обращаться к хранилищу, у него должны быть отдельные методы

DyndaNE commented on src/main/java/ru/yandex/practicum/filmorate/service/FilmService.java:
```diff
@@ -0,0 +1,66 @@
+
+    public InMemoryFilmStorage getFilmStorage() {
+        return filmStorage;
+    }
```
такого метода быть не должно, твой сервис внутри методов обращается к хранилищу, которое должно быть внедрено в этот класс через конструктор

ArsenyIshtubaev committed:
-fix:
Спринт9 (2)
- исправить внедрение зависимостей, сделать через конструктор
- убрать взаимодейтвие контроллера напрямую с хранилищем

ArsenyIshtubaev committed:
-fix:
Спринт9 (2)
- исправить внедрение зависимостей, сделать через конструктор
- убрать взаимодейтвие контроллера напрямую с хранилищем

DyndaNE commented on src/main/java/ru/yandex/practicum/filmorate/service/FilmService.java:
```diff
@@ -0,0 +1,83 @@
+@Service
+public class FilmService {
+
+    private final InMemoryFilmStorage filmStorage;
```
тут должен быть интерфейс, а не его реализация и в конструкторе соответственно тоже

DyndaNE commented on src/main/java/ru/yandex/practicum/filmorate/service/UserService.java:
```diff
@@ -0,0 +1,113 @@
+@Service
+public class UserService {
+
+    private final InMemoryUserStorage userStorage;
```
тут аналогично должен быть интерфейс, а не его реализация и в конструкторе соответственно тоже

ArsenyIshtubaev committed:
-fix:
Спринт9 (3)
- сервисы должны зависеть от интерфейсов классов-хранилищ
